### PR TITLE
fixing issue where get-droppable-over incorrectly calculates clipping

### LINKED ISSUE #266: Dimension clipping does not account for mid-drag scroll correctly when multiple Droppables share a scroll container
## Bug or feature request?
Bug

### Expected behavior
Clipped dimensions account for mid-drag scrolling when multiple Droppables share a scroll container.

### Actual behavior
When multiple Droppables share a scroll container it does not appear as if the subject's scroll position is updated during a drag.

### Steps to reproduce
1. [View the multiple vertical lists stress test story.](https://react-beautiful-dnd.netlify.com/?selectedKind=multiple%20vertical%20lists&selectedStory=stress%20test&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel)
2. Lift an item with the mouse and drag it over the Zeta list.
3. While still dragging, scroll the container down then drag the item over the top of the Eta list. The app should still behave as if you're dragging over the Zeta list.

### Demo
[Video demo here.](http://i.imgur.com/7dJUplx.gifv)

### Browser version
Chrome 63.0.3239.132

### PR DESCRIPTION
**Pull request by alexreardon**
Closes #266

### CODE REVIEW

File: src/state/dimension.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ export const clip = (frame: Area, subject: Spacing): ?Area => {
     left: Math.max(subject.left, frame.left),
   });
 
-  if (result.width < 0 || result.height < 0) {
+  if (result.width <= 0 || result.height <= 0) {
```
Comment (alexreardon) [inline src/state/dimension.js]: A little cleanup

File: src/state/get-droppable-over.js [hunk 2]
```diff
@@ -26,26 +26,35 @@ const getRequiredGrowth = memoizeOne((
...
+
+    return requiredGrowth;
+  };
+
   const dimensions: DraggableDimension[] = getDraggablesInsideDroppable(droppable, draggables);
 
+  // Droppable is empty
   if (!dimensions.length) {
-    return null;
+    const existingSpace: number = droppable.page.withMargin[droppable.axis.size];
```
Comment (alexreardon) [inline src/state/get-droppable-over.js]: While writing tests I came across this: we should add space to empty `Droppables` if required. I also added some tests for this

File: src/state/get-droppable-over.js [hunk 3]
```diff
@@ -92,7 +101,8 @@ const getClippedAreaWithPlaceholder = ({
   }
 
   const isClippedByFrame: boolean = subject[droppable.axis.size] !== frame[droppable.axis.size];
-  const subjectWithGrowth = getWithGrowth(subject, requiredGrowth);
+
+  const subjectWithGrowth = getWithGrowth(clipped, requiredGrowth);
```
Comment (alexreardon) [inline src/state/get-droppable-over.js]: this line is the fix. Rather than adding the growth to the initial subject, we add it to the updated clipped area

**Comment by jaredcrowe** [comment]
no bueno dragging to the bottom of Alpha now dude :(

**Comment by jaredcrowe** [comment]
as discussed IRL we're not concerned with supporting the case of frames growing along the main axis when the frame scrolls on the cross axis.

**Pull request merged**